src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_SIZE = 10;
+
+    private Object[] elements;
```

Comment:
You must use the parameterized type `T`
```suggestion
    private T[] elements;
```

Reply:
Is it a new requirement?
No word on it nor within README nor the comments.
Please advise. Thanks!

Create your own ArrayList class which implements the List interface. Override the methods required.

Make your ArrayList behave the same way as java.util.ArrayList (it should have the default capacity, the internal array should grow 1.5 times when it is full, etc.).

Throw custom ArrayListIndexOutOfBoundsException in case the index passed to any of the methods is invalid. Throw NoSuchElementException in remove(T element) if there is no such element present.

https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html

Reply:
Anyway, the "elements" attribute is not shared beyond the class implementation.

Reply:
Or... do you want me to play with reflection? np.

Reply:
Ok, Element class to hold value added.
Does it look better? ;-)

Reply:
See another version based on reflection.
Did you really want it to be that way? ;-)